### ISSUE TITLE
Speedtest.net Beta not loading

### ISSUE THREAD

**Issue by tcpun** [brave/browser-laptop#979]
The beta speedtest.net is not working until I disabled the adblock

![image](https://cloud.githubusercontent.com/assets/1812168/13484413/7dddc8e4-e138-11e5-8c05-182ec42845a9.png)

**Comment by luixxiul** [comment]
> Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'none'". Either the 'unsafe-inline' keyword, a hash ('sha256-UimqYw5YPuHeMthxkkMXNzmnALq2FS054tBtyrBqgPg='), or a nonce ('nonce-...') is required to enable inline execution.

Is there any way of beginning test without "Allow Ads and Tracking"?

CC: @diracdeltas

**Comment by diracdeltas** [comment]
@luixxiul please test sites with noscript disabled unless the original reporter says they have noscript enabled. that will make the 'Refused to execute inline script' errors go away.

**Comment by diracdeltas** [comment]
http://beta.speedtest.net/ is working for me with adblock enabled. can be closed?

**Comment by luixxiul** [comment]
On 0.9.6 allowing scripts I got this one and the site doesn't work. Allowing 3rd party cookies changed nothing.

````
(index):271 Uncaught ReferenceError: require is not defined(anonymous function) @ (index):271
(index):22 GET http://www.budgetedbauer.com/fs6goo net::ERR_BLOCKED_BY_CLIENTt.exports @ (index):22(anonymous function) @ (index):23r @ (index):23n @ (index):22
(index):1 (WEB_PAGE context) extensions::ipc_utils:35: Uncaught TypeError: Error processing argument at index 0, conversion failure from null{TypeError: Error processing argument at index 0, conversion failure from null
    at TypeError (native)
    at EventEmitter2.<anonymous> (extensions::ipc_utils:35:22)
    at EventEmitter2.emit (extensions::event_emitter:43:20)
    at EventEmitter2.ipcRenderer.emit (extensions::ipc_utils:31:40)}
VM70:133 blocking canvas read Object {obj: "HTMLCanvasElement.prototype", prop: "toDataURL", scriptUrl: "http://beta.speedtest.net/"}
about:blank:1 Refused to display 'http://c.speedtest.net/javascript/speedtest-main.js?p=0KIGC1funslM&r=6330cH…X%26j%3DGR&s=C330310P2JbV1g_dFtRP2RbTlZaWlZUVFRUWlVTWVRWVVFXVVM_aFsl%3D&b=' in a frame because it set 'X-Frame-Options' to 'SAMEORIGIN'.
(index):22 Resource interpreted as Document but transferred with MIME type application/javascript: "http://c.speedtest.net/javascript/speedtest-main.js?p=0KIGC1funslM&r=6330cH…X%26j%3DGR&s=C330310P2JbV1g_dFtRP2RbTlZaWlZUVFRUWlVTWVRWVVFXVVM_aFsl%3D&b=".s @ (index):22o @ (index):22(anonymous function) @ (index):22o @ (index):22(anonymous function) @ (index):23i @ (index):23(anonymous function) @ (index):23(anonymous function) @ (index):23i @ (index):23(anonymous function) @ (index):23(anonymous function) @ (index):22(anonymous function) @ (index):23
speedtest-main.js?p=0KIGC1funslM&r=6740PCZ5eVB7Zk1gW2FaYWVYYmNTeE9YW2FQUl5bWmB4WlFgeVtbV1hNeFZf&q=2…:5 Failed to execute 'write' on 'Document': It isn't possible to write into a document from an asynchronously-loaded external script unless it is explicitly opened.
````
![](https://cloud.githubusercontent.com/assets/3362943/15309188/8b19408e-1c1f-11e6-9fee-3c85a75d86b4.jpg)

**Comment by diracdeltas** [comment]
try again after allowing canvas fingerprinting in about:preferences

**Comment by luixxiul** [comment]
Hm, still doesn't work :-(

**Comment by tcpun** [comment]
@diracdeltas it will load the page, but it won't start the test - nothing would happen if I clicked Begin Test

![untitled](https://cloud.githubusercontent.com/assets/1812168/15310040/dc19e976-1c1e-11e6-94d0-c93a4981cc27.jpg)
![untitled2](https://cloud.githubusercontent.com/assets/1812168/15310072/20948e8a-1c1f-11e6-9bea-00b3bdc2fce8.jpg)

**Comment by luixxiul** [comment]
Same here on master branch. allowing canvas fingerprinting, ads and tracking only works.

**Comment by srirambv** [comment]
http://beta.speedtest.net works with disabling FP on the latest version. Closing #979

**Comment by srirambv** [comment]
Reopening the issue from #8772

**Comment by Jacalz** [comment]
I can get the test to start by "allowing ads and trackers" and fingerprinting, seems like I need to have fingerprinting and adblock disabled to start the test on my windows 10 pc running 0.15.2 :wink: 
This is really annoying and it need a fix asap :+1:

**Comment by pmo** [comment]
I have this issue aswell when adblock is enabled on Brave 0.15.2 rev af7ef42. Tested on Google Chrome using the same blocklists, works fine there.

**Comment by Jacalz** [comment]
This is really annoying, @lukemulks really needs to take a look at it when he gets back from vacation :+1:

**Comment by lukemulks** [comment]
Will investigate!

**Comment by lukemulks** [comment]
I've been able to test this (US: Dublin, CA), and the beta.speedtest.net site is working OK with me with default settings: Ad blocking enabled, tracking protection enabled. 

In this version of Brave: 
```
Name	Version
Brave	0.15.304
rev	b872523
```

![inkedbrave-speedtest-05222017-completed_li](https://cloud.githubusercontent.com/assets/8979240/26325953/eeb02850-3eed-11e7-9e08-e3725b5ea23a.jpg)

Also, my console from the successful test: 
![brave-speedtest-05222017-completed-console](https://cloud.githubusercontent.com/assets/8979240/26326096/a2f392ac-3eee-11e7-838e-40a5d5c42c9e.PNG)

The test *is not* working for me with Fingerprinting Protection enabled, but is working for me with default Brave settings enabled. 

I'd suggest users that were/are experiencing this disable FP in the site settings (lion menu), and see if that resolves the issue (while keeping ad blocking and tracking protection enabled). 

Let us know if the issue persists, even with FP disabled.

**Comment by Jacalz** [comment]
I have the same issue with Adblock but as I said to you, will look in the new versions @lukemulks

**Comment by Jacalz** [comment]
I can confirmed that Adblock is fixed on the latest master, that is on windows 10 atleast... :+1: 
With adblock:
![image](https://cloud.githubusercontent.com/assets/25466657/26354649/d0cb6a10-3fc5-11e7-9db4-888554916e2a.png)
With fingerprinting and adblock:
![image](https://cloud.githubusercontent.com/assets/25466657/26354695/fb6dc4fc-3fc5-11e7-9858-47eedabcdbb6.png)
Can someone confirm on linux and osx?

**Comment by pmo** [comment]
Works for me in Windows10, MacOS and Linux if I disable everything.

**Comment by luixxiul** [comment]
This tweet shows that the issue has been fixed: https://twitter.com/orosseel/status/898928289775058944
The site works for me too by disabling fingerprinting protection.

Requires a QA test on 0.21 as blocking 3rd party fingerprinting will be default thanks to https://github.com/brave/browser-laptop/pull/10181.

**Comment by luixxiul** [comment]
Added `release-notes/include` just in case where other `webcompat` issues should be fixed on 0.21 thanks to the new default FP setting and some of them would be listed as examples.
